src/shared/components/plots/WaterfallPlot.tsx

```diff
@@ -0,0 +1,545 @@
+        } else {
+            return null;
+        }
+    }
+
+    @computed get plotDomain():{value:number[], order:number[]} {
+        // data extremes
+        let max = _(this.internalData).map('value').max() || 0;
+        let min = _(this.internalData).map('value').min() || 0;
+
+        return {
+            value: [min!, max!],
+            order: [1, this.internalData.length]  // return range defined by the number of samples for the x-axis
+        };
+    }
+
+    @computed get plotDomainX() {
+        if (this.props.horizontal) {
+            return this.plotDomain.value;
+        }
+        return this.plotDomain.order;
+    }
+
+    @computed get plotDomainY() {
+        if (this.props.horizontal) {
+            return this.plotDomain.order;
+        }
+        return this.plotDomain.value;
+    }
+
+    @computed get rightPadding() {
+        return RIGHT_PADDING;
+    }
+
+    @computed get svgWidth() {
+        return LEFT_PADDING + this.props.chartWidth + this.rightPadding;
+    }
+
+    @computed get svgHeight() {
+        return this.props.chartHeight;
+    }
+
+    @bind
+    private datumAccessorY(d:IBaseWaterfallPlotData) {
+        return d.value;
+    }
+
+    @bind
+    private datumAccessorBaseLine(d:IBaseWaterfallPlotData) {
+        return d.offset;
+    }
+
+    @bind
+    private datumAccessorX(d:IBaseWaterfallPlotData) {
+        return d.order;
+    }
+
+    @bind
+    private datumAccessorLabelY(d:IBaseWaterfallPlotData) {
+        return d.labely;
+    }
+
+    @bind
+    private datumAccessorLabelX(d:IBaseWaterfallPlotData) {
+        return d.labelx;
+    }
+
+    @bind
+    private datumAccessorSearchIndicatorY(d:IBaseWaterfallPlotData) {
+        return d.searchindicatory;
+    }
+
+    @bind
+    private datumAccessorSearchIndicatorX(d:IBaseWaterfallPlotData) {
+        return d.searchindicatorx;
+    }
+
+    @computed get size() {
+        const highlight = this.props.highlight;
+        const size = this.props.size;
+        // need to regenerate this function whenever highlight changes in order to trigger immediate Victory rerender
+        return makePlotSizeFunction(highlight, size);
+    }
+
+    private tickFormat(t:number, ticks:number[], logScaleFunc:IAxisLogScaleParams|undefined) {
+        if (logScaleFunc && !this.props.useLogSpaceTicks) {
+            t = logScaleFunc.fInvLogScale(t);
+            ticks = ticks.map(x=>logScaleFunc.fInvLogScale(x));
+        }
+        return tickFormatNumeral(t, ticks);
+    }
+
+    @bind
+    private tickFormatX(t:number, i:number, ticks:number[]) {
+        if (this.props.horizontal) {
+            return this.tickFormat(t, ticks, this.props.log);
+        }
+        return undefined;
+    }
+
+    @bind
+    private tickFormatY(t:number, i:number, ticks:number[]) {
+        if (this.props.horizontal) {
+            return undefined;
+        }
+        return this.tickFormat(t, ticks, this.props.log);
+    }
+
+    @computed get internalData() {
```

Comment:
more descriptive name please

Reply:
What do you think of the name `waterfallPlotdata`?